config/crd/bases/infrastructure.cluster.x-k8s.io_openstackclusters.yaml

```diff
@@ -5240,9 +5240,33 @@ spec:
                   type: string
                 type: array
                 x-kubernetes-list-type: set
-              externalNetworkId:
-                description: ExternalNetworkID is the ID of an external OpenStack
-                  Network. This is necessary to get public internet to the VMs.
+              externalNetwork:
+                description: ExternalNetwork is the OpenStack Network to be used to
+                  get public internet to the VMs.
+                properties:
+                  description:
+                    type: string
+                  id:
+                    type: string
+                  name:
+                    type: string
+                  notTags:
+                    type: string
+                  notTagsAny:
+                    type: string
+                  projectId:
+                    type: string
+                  tags:
+                    type: string
+                  tagsAny:
+                    type: string
+                type: object
+              externalNetworkType:
```

Comment:
There is pattern for naming the API fields that mean to disable something, which I think is what this  does right?
might be worth to rename it to `disableExternalNetwork`.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/api/v1alpha8/openstackcluster_types.go#L91